src/components/Popups/PopupDrawingToolbar.tsx

```diff
@@ -50,16 +54,28 @@ const PopupDrawingToolbar = ({
     canRedo,
     canUndo,
     className,
+    drawnPathGroups,
     onDelete,
     onRedo,
     onReply,
     onUndo,
     reference,
 }: Props): JSX.Element => {
     const intl = useIntl();
+    const popupRef = React.useRef<PopupBase>(null);
+    const prevDrawnPathGroups = usePrevious(drawnPathGroups);
+
+    React.useEffect(() => {
+        const { current: popup } = popupRef;
+
+        if (popup?.popper && prevDrawnPathGroups && drawnPathGroups.length !== prevDrawnPathGroups.length) {
+            popup.popper.update();
+        }
+    }, [popupRef, drawnPathGroups, prevDrawnPathGroups]);
```

Comment:
This seems a bit too involved for `PopupDrawingToolbar`. Could we move this logic up to `DrawingAnnotations` since it has the props/knowledge on when to update the location? We could forward the ref to the `PopupBase` here and in DrawingAnnotations use that ref to do the update based on `drawnPathGroups`